src/main/scripts/prepare_release.py

```diff
@@ -0,0 +1,183 @@
+import git
+import logging as log
+import os
+import re
+import subprocess
+import sys
+from argparse import ArgumentParser
+from datetime import datetime
+from ruamel.yaml import YAML
+from tempfile import mkstemp
+
+products = ["bamboo", "bamboo-agent",
+            "bitbucket", "confluence", "crowd", "jira"]
+prodbase = "src/main/charts"
+
+def get_chart_versions(path=prodbase):
+    versions = {}
+    for prod in products:
+        proddir = f"{path}/{prod}"
+        chartfile = f"{proddir}/Chart.yaml"
+
+        with open(chartfile, 'r') as chart:
+            yaml = YAML()
+            chartyaml = yaml.load(chart)
+
+        versions[prod] = {}
+        versions[prod]['kubeVersion'] = chartyaml['kubeVersion']
+        versions[prod]['appVersion'] = chartyaml['appVersion']
+        versions[prod]['version'] = chartyaml['version']
+
+        log.info(f"Product {prod} versions: {versions[prod]['kubeVersion']}, {versions[prod]['appVersion']}")
+
+    return versions
+
+
+def gen_changelog(product, path):
+    repo = git.Repo(path)
+    cli = git.Git(path)
+    tags = sorted(repo.tags, key=lambda t: t.commit.committed_datetime)
+
+    lasttag = tags[-1]
+    tagver = re.sub(r'^[^-]+-', '', lasttag.name)
+    log.info(f'Generating {product} changelog since {tagver}')
+
+    # git log will pick up commits in src/main/charts/$product directory only
+    changelog = cli.log(f'{lasttag}..main', "--", prodbase + '/' + product, graph=True, pretty='format:%s',
+                        abbrev_commit=True, date='relative', )
+
+    changelog = changelog.split('\n')
+    # we don't need pre-release commits that update chart.yamls and changelogs
+    # this pattern is based on the commit message from a GitHub action that prepares Helm release
+    pattern = re.compile(r'^\* Prepare release [0-9].{1,4}')
+    filtered_changelog = list(filter(lambda x: not pattern.match(x), changelog))
+
+    if len(filtered_changelog) == 0 or filtered_changelog.count(''):
+        # It is possible that there are no commits to the Helm chart,
+        # in this case we just write a generic git log
+        log.info(f'No commits to {product} Helm chart found')
+        default_git_log = '* Update Helm chart version'
+        filtered_changelog = [default_git_log]
+
+    # remove Jira keys from commit messages. This substitution assumes the commit message may have the following format:
+    # CLIP-1234: Here is my message
+    # DCCLIP-1234: Here is my message
+    sanitized_changelog = map(lambda c: re.sub(r'(CLIP|DCCLIP)-[0-9]{1,5}(?![0-9]): ', '', c), filtered_changelog)
+    return list(dict.fromkeys(sanitized_changelog))
+
+
+def format_changelog_yaml(changelog):
+    # The artifacthub annotations are a single string, but formatted
+    # like YAML. Replacing the leading '*' with '-'  and wrapping
+    # strings in double quotes should be sufficient.
+    sanitized_changelog = []
+    for string in changelog:
+        string = string + '"'
```

Comment:
Should it be from both sides so it works for the first one as well?

Reply:
@nanux so the original string is `* I am a string`, and first it is transformed into `* I am a string"` and then to `- "I am a string"` to fix the yaml annotations.